src/components/proposal/ProposalContainer.tsx

```diff
@@ -38,30 +37,30 @@ const proposalReducer = (
   action: Event
 ) => {
   switch (action.type) {
-    case EventType.PROPOSAL_CREATED:
-    case EventType.PROPOSAL_LOADED:
-      const proposal: Proposal = action.payload.proposal;
+    case 'PROPOSAL_CREATED':
+    case 'PROPOSAL_LOADED':
+      const proposal = action.proposal;
       draftState.isDirty = false;
       draftState.questionaryId = proposal.questionaryId;
       draftState.proposal = proposal;
       draftState.steps = proposal.questionary?.steps || [];
       draftState.templateId = proposal.questionary?.templateId || 0;
       break;
-    case EventType.PROPOSAL_MODIFIED:
+    case 'PROPOSAL_MODIFIED':
```

Comment:
Not sure why instead of the enums it is better to use strings to compare the action type. I think it was better to use the enum to prevent typos and some other mistakes.